castai/resource_workload_scaling_policy.go

```diff
@@ -0,0 +1,341 @@
+package castai
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/castai/terraform-provider-castai/castai/sdk"
+)
+
+var (
+	k8sNameRegex = regexp.MustCompile("^[a-z0-9A-Z][a-z0-9A-Z._-]{0,61}[a-z0-9A-Z]$")
+)
+
+func resourceWorkloadScalingPolicy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceWorkloadScalingPolicyCreate,
+		ReadContext:   resourceWorkloadScalingPolicyRead,
+		UpdateContext: resourceWorkloadScalingPolicyUpdate,
+		DeleteContext: resourceWorkloadScalingPolicyDelete,
+		CustomizeDiff: resourceWorkloadScalingPolicyDiff,
+		Importer: &schema.ResourceImporter{
+			StateContext: workloadScalingPolicyImporter,
+		},
+		Description: "Manage workload scaling policy. Scaling policy [reference](https://docs.cast.ai/docs/woop-scaling-policies)",
+		Schema: map[string]*schema.Schema{
+			FieldClusterID: {
```

Comment:
nit: they could also set ConfidenceThreshold, but it is probably not important :D